discovery-client/src/main/java/io/micronaut/discovery/vault/config/client/v1/VaultConfigConfigurationClientV1.java

```diff
@@ -0,0 +1,189 @@
+
+                        String vaultSourceName = getVaultSourceName(activeNames, pairVaultResponse.getLeft());
+                        Map<String, Object> vaultResponseData = vaultResponse.getData();
+
+                        synchronized (sourceCount) {
+                            sourceCount.getAndIncrement();
+                        }
+
+                        if (!CollectionUtils.isEmpty(vaultResponseData)) {
+                            if (LOG.isInfoEnabled()) {
+                                LOG.info("Obtained property source from Vault, source={}", vaultSourceName);
+                            }
+                            emitter.onNext(PropertySource.of(vaultSourceName,
+                                    vaultResponseData, Integer.MAX_VALUE - activeNamesList.indexOf(
+                                            pairVaultResponse.getLeft())));
+                        }
+
+                        //if all items have been processed, emit onComplete
+                        if (sourceCount.get() == configurationValuesList.size()) {
+                            emitter.onComplete();
+                        }
+            }, BackpressureStrategy.ERROR));
+        });
+    }
+
+    /**
+     * @param activeNames active environment names
+     * @return list of responses from vault
+     */
+    private List<PairVaultResponse> retrieveVaultProperties(Set<String> activeNames) {
+        final String applicationName = getApplicationConfiguration().getName().get();
+
+        return activeNames.stream().map(activeName -> new PairVaultResponse(activeName,
+                    activeName.equals(applicationName) ?
+                        Flowable.fromPublisher(
+                                vaultConfigClientV1.readConfigurationValues(
+                                    getVaultClientConfiguration().getBackend(), applicationName)) :
+                        Flowable.fromPublisher(
+                                vaultConfigClientV1.readConfigurationValues(
+                                    getVaultClientConfiguration().getBackend(), applicationName, activeName))
+                )).collect(Collectors.toList());
+    }
+
+    /**
+     * @param sourceCount total of property sources from vault
+     * @return the error handler to handle vault failed requests
+     */
+    private Function<Throwable, Publisher<? extends VaultResponseV1>> getErrorHandler(AtomicInteger sourceCount) {
+        return throwable -> {
+
+            synchronized (sourceCount) {
+                sourceCount.getAndIncrement();
+            }
+
+            if (throwable instanceof HttpClientResponseException) {
+                HttpClientResponseException httpClientResponseException = (HttpClientResponseException) throwable;
+                if (httpClientResponseException.getStatus() == HttpStatus.NOT_FOUND) {
+                    if (getVaultClientConfiguration().isFailFast()) {
```

Comment:
Is not having configuration in the remote source considered an error condition?

Reply:
This would be a decision design @jameskleeh. If not having a configuration in the remote considerer not an error, I can remove this condition so it does not fail.

Reply:
It isn't considered an error condition in the other implementations. To me its an expected outcome and not an error

Reply:
Sounds good to me, I will apply this behavior. Should we still maintain the `fail-fast` condition and just remove the check for `NOT_FOUND`?